schwab_api/authentication.py

```diff
@@ -88,62 +48,67 @@ def login(self, username, password, totp_secret=None):
         """
-        
-        # Log in to schwab using Playwright
-        with self.page.expect_navigation():
-            self.page.goto("https://www.schwab.com/")
+        result = asyncio.run(self._async_login(username, password, totp_secret))
+        return result
+    
+    async def _async_login(self, username, password, totp_secret=None):
+        """ This function runs in async mode to perform login.
+        Use with login function. See login function for details.
+        """
+        self.playwright = await async_playwright().start()
+        if self.browserType == "firefox":
+            self.browser = await self.playwright.firefox.launch(
+                headless=self.headless
+            )
+        else:
+            raise ValueError("Only supported browserType is 'firefox'")
 
+        user_agent = USER_AGENT + self.browser.version
+        self.page = await self.browser.new_page(
+            user_agent=user_agent,
+            viewport=VIEWPORT
+        )
+        await stealth_async(self.page)
+        
+        await self.page.goto("https://www.schwab.com/")
 
-        # Capture authorization token.
-        self.page.route(re.compile(r".*balancespositions*"), self.captureAuthToken)
+        await self.page.route(re.compile(r".*balancespositions*"), self._asyncCaptureAuthToken)
 
-        # Wait for the login frame to load
         login_frame = "schwablmslogin"
-        self.page.wait_for_selector("#" + login_frame)
+        await self.page.wait_for_selector("#" + login_frame)
 
-        self.page.frame(name=login_frame).select_option("select#landingPageOptions", index=3)
+        await self.page.frame(name=login_frame).select_option("select#landingPageOptions", index=3)
+
+        await self.page.frame(name=login_frame).click("[placeholder=\"Login ID\"]")
+        await self.page.frame(name=login_frame).fill("[placeholder=\"Login ID\"]", username)
 
-        # Fill username
-        self.page.frame(name=login_frame).click("[placeholder=\"Login ID\"]")
-        self.page.frame(name=login_frame).fill("[placeholder=\"Login ID\"]", username)
-        
-        # Add TOTP to password
         if totp_secret is not None:
             totp = pyotp.TOTP(totp_secret)
             password += str(totp.now())
 
-        # Fill password
-        self.page.frame(name=login_frame).press("[placeholder=\"Login ID\"]", "Tab")
-        self.page.frame(name=login_frame).fill("[placeholder=\"Password\"]", password)
+        await self.page.frame(name=login_frame).press("[placeholder=\"Login ID\"]", "Tab")
+        await self.page.frame(name=login_frame).fill("[placeholder=\"Password\"]", password)
 
-        # Submit
         try:
-            with self.page.expect_navigation():
-                self.page.frame(name=login_frame).press("[placeholder=\"Password\"]", "Enter")
+            await self.page.frame(name=login_frame).press("[placeholder=\"Password\"]", "Enter")
```

Comment:
I am also experiencing login issues today.
It seems that the url to place trades has changed, at least on my end.
I have made a change to the login to ensure more robustness by looking for the keywords "app/trade" in the url instead of an exact url.